File: apps/alfred/schemas/threads.py
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class ThreadCreate(BaseModel):
    kind: str = Field(..., min_length=1)
    title: str | None = None
    user_id: int | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)


class Thread(BaseModel):
    id: str
    kind: str
    title: str | None = None
    user_id: int | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime | None = None
    updated_at: datetime | None = None


class ThreadMessageCreate(BaseModel):
    role: str = Field(..., min_length=1)
    content: str | None = None
    data: dict[str, Any] = Field(default_factory=dict)


class ThreadMessage(BaseModel):
    id: str
    thread_id: str
    role: str
    content: str | None = None
    data: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime | None = None
    updated_at: datetime | None = None


__all__ = [
    "Thread",
    "ThreadCreate",
    "ThreadMessage",
    "ThreadMessageCreate",
]

File: frontend/lib/api/types/threads.ts
export type Thread = {
  id: string;
  kind: string;
  title?: string | null;
  user_id?: number | null;
  metadata: Record<string, unknown>;
  created_at?: string | null;
  updated_at?: string | null;
};

export type ThreadMessageRole = string;

export type ThreadMessage = {
  id: string;
  thread_id: string;
  role: ThreadMessageRole;
  content?: string | null;
  data: Record<string, unknown>;
  created_at?: string | null;
  updated_at?: string | null;
};

export type CreateThreadRequest = {
  kind: string;
  title?: string | null;
  user_id?: number | null;
  metadata?: Record<string, unknown>;
};

export type AppendThreadMessageRequest = {
  role: ThreadMessageRole;
  content?: string | null;
  data?: Record<string, unknown>;
};
